pennylane/spin/lattice.py

```diff
@@ -0,0 +1,196 @@
+# Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains functions and classes to create a
+:class:`~pennylane.spin.lattice` object. This object stores all
+the necessary information about a lattice.
+"""
+import itertools
+
+import numpy as np
+from scipy.spatial import cKDTree
+
+# pylint: disable=too-many-arguments, too-many-instance-attributes
+# pylint: disable=use-a-generator, too-few-public-methods
+
+
+class Lattice:
+    r"""Constructs a Lattice object.
+
+    Args:
+       L: Number of unit cells in each direction.
+       unit_cell: Primitive vectors for the lattice.
+       basis: Initial positions of spins.
+       boundary_condition: defines boundary conditions, boolean or series of bools with dimensions same as L.
+       neighbour_order: Range of neighbours a spin interacts with.
```

Comment:
I thought `neighbour_order` defines interaction up to the given order. 🤔

Reply:
I don't think I follow, it defines the interactions, as in nearest neighbour is 1, then. next nearest neighbour will be 2, and so on.

Reply:
Yep! I don't think that is quite covered in the description yet. Could we make it clearer there? 😄